File: Pow(x,n).py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        temp=0
        if n==0:
            return 1.0
        if n<0:
            return self.myPow(1/x,-n)
        temp=self.myPow(x,n//2)
        if n%2==0:
            return temp*temp
        else:
            if n>0:return x*temp*temp

File: Algorithms/Pow(x,n).java
class Solution {
    public double myPow(double x, long n) {
        if (n==0 || x==1){
            return 1;
        }
        if (n<0){
            return myPow(1/x,-n);
        }
        double temp=myPow(x,n/2);
        if (n%2==0){
            return temp*temp;
        }else{
            return x*temp*temp;
        }
    }
}